storage/range_command.go

```diff
@@ -229,43 +229,66 @@ func (r *Range) Scan(batch engine.Engine, args proto.ScanRequest) (proto.ScanRes
 
 // EndTransaction either commits or aborts (rolls back) an extant
 // transaction according to the args.Commit parameter.
-func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args proto.EndTransactionRequest) (proto.EndTransactionResponse, error) {
+func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args proto.EndTransactionRequest) (proto.EndTransactionResponse, []proto.Intent, error) {
 	var reply proto.EndTransactionResponse
+	abort := func(err error) (proto.EndTransactionResponse, []proto.Intent, error) {
```

Comment:
This is an odd pattern; I'd rather just `return reply, nil, util.Errorf(...)` at each call site.